synapse/datamodel.py

```diff
@@ -116,6 +116,12 @@ def __init__(self, modl, form, name, typedef, info):
             form.setProp(name, self)
             self.modl.propsbytype[self.type.name].append(self)
 
+        if self.info.get('deprecated') or self.type.deprecated:
+            async def depfunc(node, oldv):
+                mesg = f'The property {self.full} is deprecated or using a deprecated type and will be removed in 3.0.0'
```

Comment:
Would we want a "once" type filter (per snap?) to not send a million of these?

Reply:
( and at other spots obvs )

Reply:
maybe a deprwarn(name) method on the snap that can track if it's been sent already or something

Reply:
sure. we can easily do something like that on the snap that gates a given name to only warn it once. the idea here was that we would warn whenever a deprecated form or property was used.

Reply:
yep, completely agreed